### ISSUE TITLE
`alwaysIgnoredGlobs` prevents using stylelint for files under node_modules

### ISSUE THREAD

**Issue by dmp42** [stylelint/stylelint#2523]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Any file under a directory listed in `alwaysIgnoredGlobs` will be ignored (https://stylelint.io/user-guide/configuration/#stylelintignore).

Understood this was introduced to fix #2149 so that imports referencing dependencies would not be linted, but this is effectively preventing using stylelint against an explicit file path containing for eg "node_modules".

There are scenarios (typically Alle - see pouchdb) where several modules in the same git repo happen to be developed under a directory containing `node_modules` and where it is desirable to be able to use stylelint.

I guess a solution may be to make `alwaysIgnoredGlobs` configurable?

As a side-note, `bower_components` is also hardcoded in `alwaysIgnoredGlobs` and this will fail to work as intended in case the project uses a non-standard location via `.bowerrc`.

> Which rule, if any, is this issue related to?

None.

> What CSS is needed to reproduce this issue?

None.

> What stylelint configuration is needed to reproduce this issue?

None.

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config .stylelintrc whatever/node_modules/whatever/whatever.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

The file should be linted.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Error: whatever/node_modules/whatever/whatever.css does not match any files
    at globby.then.filePaths (node_modules/stylelint/lib/standalone.js:129:33)
```

**Comment by dmp42** [comment]
In case anyone stumbles here for the same reason, here is a terrible way to workaround this without forking.

```
  const Module = require('module');
  const originalCompile = Module.prototype._compile;

  const terribleHack = function (content, filename, done) {
    if (filename.includes('/stylelint/lib/alwaysIgnoredGlobs.js')) {
      Module.prototype._compile = originalCompile;
      return done('module.exports = [];');
    }
    return done();
  };

  Module.prototype._compile = function (content, filename) {
    const self = this;
    terribleHack(content, filename, function(override) {
      originalCompile.call(self, override || content, filename);
    });
  };

  const style = require('gulp-stylelint');
  // Go on with your life after that
```

**Comment by jeddy3** [comment]
@dmp42 Thanks for report and for using the template.

I believe this has been recently addressed in https://github.com/stylelint/stylelint/pull/2464 and will be released in `8.0.0`. 

Would you like to help us by testing the `v8` branch on your codebase? It'd be great to have confirmation that this specific issue will be resolved in the next major release.

Until the release of `8.0.0` (there are a [few outstanding issues to go](https://github.com/stylelint/stylelint/milestone/23)) I suspect that your hack is likely the only way to work around this limitation.

**Comment by dmp42** [comment]
Hey @jeddy3 

Thanks for the very fast turnaround. Sorry that I missed #2464 

Just tested the v8 branch, and confirming this works as expected and fixes this!

Closing, thanks a lot!
